### PR TITLE
Apache couchdb

### PR DESCRIPTION
**Pull request by JD2344**
Continuing on from Sofia's PR. This is now in a working state in SecGen. 

The only thing i will need to review is the "port". Seems the exploit runs from a separate requirement software which is hosted on a different port. I will need to look into this further to make it dynamic.

So far exploit runs as user and correctly runs through everything.

### CODE REVIEW

**Comment by JD2344** [comment]
This is working and exploits within user context. 

Cant work out why i cant seem to change the port on the erlang daemon so for now is always the same port.

File: modules/vulnerabilities/unix/http/apache_couchdb/secgen_metadata.xml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  </description>
+
+  <type>remote</type>
+  <privilege>user_rwx</privilege>
+  <access>remote</access>
+  <platform>linux</platform>
+  <difficulty>low</difficulty>
+
+  <read_fact>port</read_fact>
+  <read_fact>known_username</read_fact>
```
Comment (cliffe) [inline modules/vulnerabilities/unix/http/apache_couchdb/secgen_metadata.xml:21]: These listed parameters have to match the ones for default inputs below. If you read_fact `known_username`, then it should/can also have a default input.

File: modules/vulnerabilities/unix/http/apache_couchdb/templates/local.ini.erb [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+;example.com = /database/
+
+; To create an admin account uncomment the '[admins]' section below and add a
+; line in the format 'username = password'. When you next start CouchDB, it
+; will change the password to a hash (so that your passwords don't linger
+; around in plain-text files). You can add more admin accounts with more
+; 'username = password' lines. Don't forget to restart CouchDB after
+; changing this.
+[admins]
+<%= @username %> = <%= @password %>
```
Comment (cliffe) [inline modules/vulnerabilities/unix/http/apache_couchdb/templates/local.ini.erb:106]: Is this intended to be a password that's leaked to them on completion? Sounds like it's hashed, so might pose a problem if the thing passed to `passwords_to_leak` is not an easily crackable password

Comment (cliffe) [reply modules/vulnerabilities/unix/http/apache_couchdb/templates/local.ini.erb:106]: Is this line required for the exploit to work?
`<%= @username %> = <%= @password %>`

Comment (cliffe) [reply modules/vulnerabilities/unix/http/apache_couchdb/templates/local.ini.erb:106]: Ok, I'm going to merge this, and leave this in, but tweak the metadata.

**Pull request merged**